src/verify_cert.rs

```diff
@@ -426,6 +429,19 @@ fn check_basic_constraints(
     }
 }
 
+/// Additional context for the `RequiredEkuNotFoundContext` error variant.
+///
+/// The contents of this type depend on whether the `alloc` feature is enabled.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct RequiredEkuNotFoundContext {
```

Comment:
I've fiddled with this a fair bunch. The challenge here is that most of the internals of `KeyUsage` are private API.

* `KeyUsage` (the public API surface) also represents the `Required` vs `RequiredIfPresent` dichotomy, which is relevant for `required` but not for `present` so much.
* Internally we currently use `KeyPurposeId` which uses an internal `Input` (which means the only way to make it `'static` is with a '&`static [u8]` which isn't feasible here).
* We might want to represent `EKU_CLIENT_AUTH` and `EKU_SERVER_AUTH` in a way that are actually usable with the public API, which is `KeyUsage::client_auth()` and `KeyUsage::server_auth()`, and represent other OIDs in a somewhat readable representation.